### PR TITLE
STYLE: fix pylint redefined-oute-name warnings (pandas-dev#49656)

### PR DESCRIPTION
**Pull request by dhrubajyoti-giri**
Fixed warnings for the following files :-
- pandas/core/internals/concat.py
- pandas/core/reshape/merge.py
- pandas/io/json/_json.py

### CODE REVIEW

File: pandas/core/internals/concat.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-import copy
+import copy as _copy_
```
Comment (MarcoGorelli) [inline pandas/core/internals/concat.py:3]: how about `import copy as stdlib_copy`?

File: pandas/core/reshape/merge.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 """
 from __future__ import annotations
 
-import copy
+import copy as _copy_
```
Comment (MarcoGorelli) [inline pandas/core/reshape/merge.py:6]: same comment as above

File: pandas/io/json/_json.py [hunk 3]
```diff
@@ -912,7 +912,7 @@ def read(self) -> DataFrame | Series:
         self.close()
         return obj
 
-    def _get_object_parser(self, json) -> DataFrame | Series:
+    def _get_object_parser(self, json_object) -> DataFrame | Series:
```
Comment (MarcoGorelli) [inline pandas/io/json/_json.py:915]: if you're no longer import `json`, then can this argument stay the same?

File: pandas/io/json/_json.py [hunk 4]
```diff
@@ -1009,7 +1009,7 @@ class Parser:
 
     def __init__(
         self,
-        json,
+        json_object,
```
Comment (MarcoGorelli) [inline pandas/io/json/_json.py:1012]: same as above

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for working on this!

File: pandas/core/internals/concat.py (hunk 1 above)
Comment (giridhrubajyoti2002) [reply pandas/core/internals/concat.py:3]: right, i am working on that.

**Comment by MarcoGorelli** [comment]
will need to fixup the pre-commit checks, but apart from that looks good

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
@giridhrubajyoti2002 can you run
```
pre-commit run --files pandas/core/internals/concat.py
```
(and similarly for the other files you've modified) please?

**Comment by MarcoGorelli** [comment]
closing as stale, thanks anyway for your PR!

**Pull request closed without merging**